### PR TITLE
Improved deferrals usage, `CoreApplication.Entered/LeavingBackground`

### LINKED ISSUE #7711: `LeavingBackgroundEventArgs` and `EnteredBackgroundEventArgs` `GetDeferral` should not throw
## What would you like to be added:

A dummy implementation of `GetDeferral` for `LeavingBackgroundEventArgs` and `EtneredBackgroundEventArgs` to avoid exceptions when used.

As the proper defer behavior cannot be achieved on all platforms, there should still be a not implemented warning and descriptive comment.

## Why is this needed:

Usage simplicity.

## For which Platform:

All

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #7711 

## PR Type

What kind of change does this PR introduce?

- Feature
- Refactoring (no functional changes, no api changes)

## What is the current behavior?

- `GetDeferral` on `Entered`/`Leaving` background throws.
- Usage of `DeferralManager` is complex in code
- `CoreApplication.Entered/LeavingBackground` events are never raised

## What is the new behavior?

- Does not throw (but does not guarantee the app will wait for deferral completion on some targets.
- Simplified the usage of `DeferralManager` in the codebase.
- `CoreApplication.Entered/LeavingBackground` events are raised along with `Application` counterparts

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

**Review by davidjohnoliver — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Application.Android.cs [hunk 1]
```diff
@@ -35,10 +35,8 @@ public Application()
 			Resuming?.Invoke(null, null);
 		}
 
-		partial void OnSuspendingPartial()
-		{
-			Suspending?.Invoke(this, new Windows.ApplicationModel.SuspendingEventArgs(new Windows.ApplicationModel.SuspendingOperation(DateTime.Now.AddSeconds(30))));
-		}
+		private SuspendingOperation CreateSuspendingOperation() =>
+			new SuspendingOperation(DateTimeOffset.Now.AddSeconds(30), null);
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.Android.cs]: The value was actually not correct, updated accordingly

**Pull request merged**